### PR TITLE
Move FleetAutoscaling to autoscaling.agones.dev group

### PR DESCRIPTION
**Pull request by markmandel**
This is a pure refactor, to move FleetAutoscaling to the group outlined in #703, where it will stay.

Did this one first, since it seemed like the easiest one to tackle.

### CODE REVIEW

File: pkg/apis/autoscaling/register.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (aLekSer) [inline pkg/apis/autoscaling/register.go]: nit as per #699 pull request and in other files.
```suggestion
// Copyright 2019 Google LLC All Rights Reserved.
```

File: pkg/apis/autoscaling/v1alpha1/register.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (aLekSer) [inline pkg/apis/autoscaling/v1alpha1/register.go]:
```suggestion
// Copyright 2019 Google LLC All Rights Reserved.
```

File: pkg/apis/autoscaling/v1alpha1/doc.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (aLekSer) [inline pkg/apis/autoscaling/v1alpha1/doc.go]:
```suggestion
// Copyright 2019 Google LLC All Rights Reserved.
```

File: pkg/apis/autoscaling/v1alpha1/fleetautoscaler.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018 Google LLC All Rights Reserved.
+// Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (aLekSer) [inline pkg/apis/autoscaling/v1alpha1/fleetautoscaler.go]:
```suggestion
// Copyright 2019 Google LLC All Rights Reserved.
```

File: site/content/en/docs/Getting Started/create-fleetautoscaler.md [hunk 5]
```diff
@@ -30,9 +30,17 @@ kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/agones/{{
 
 You should see a successful output similar to this :
 
+{{% feature expiryVersion="0.10.0" %}}
```
Comment (aLekSer) [inline site/content/en/docs/Getting Started/create-fleetautoscaler.md]: I think it should expiry at the same moment as new one published
```suggestion
{{% feature expiryVersion="0.11.0" %}}
```

File: site/content/en/docs/Getting Started/create-fleetautoscaler.md [hunk 6]
```diff
@@ -44,6 +52,7 @@ kubectl describe fleetautoscaler simple-udp-autoscaler
 
 It should look something like this:
 
+{{% feature expiryVersion="0.10.0" %}}
```
Comment (aLekSer) [inline site/content/en/docs/Getting Started/create-fleetautoscaler.md]:
```suggestion
{{% feature expiryVersion="0.11.0" %}}
```

File: pkg/apis/autoscaling/register.go (hunk 1 above)
Comment (markmandel) [reply pkg/apis/autoscaling/register.go]: Ah - excellent catch. Will fix!

File: site/content/en/docs/Getting Started/create-fleetautoscaler.md (hunk 5 above)
Comment (markmandel) [reply site/content/en/docs/Getting Started/create-fleetautoscaler.md]: Yeees. Good point.

File: pkg/apis/autoscaling/register.go (hunk 1 above)
Comment (markmandel) [reply pkg/apis/autoscaling/register.go]: Done!

**Comment by markmandel** [comment]
Kubernetes is being super, super weird - looks like it caching the discovery path for the fleetautoscaler to the old group. Doing a full `helm delete agones -purge` to see if it solves the issue.

**Comment by markmandel** [comment]
Next attempt - upgrade to 1.11.10, and update the nodes too.

**Comment by markmandel** [comment]
it was not k8s, it was a small thing in my code. Worked it out, will published again when this is ready for review (it's not right now).

**Comment by markmandel** [comment]
This is now ready for review :+1: 

(That was fun!)

File: test/e2e/fleetautoscaler_test.go [hunk 7]
```diff
@@ -20,7 +20,8 @@ import (
 	"testing"
 	"time"
 
-	"agones.dev/agones/pkg/apis/stable/v1alpha1"
+	"agones.dev/agones/pkg/apis/autoscaling/v1alpha1"
```
Comment (pooneh-m) [inline test/e2e/fleetautoscaler_test.go]: Can you keep this consistent throughout the tests? Like stablev1alpha1 and autoscalingv1alpha1?

File: test/e2e/fleetautoscaler_test.go [hunk 8]
```diff
@@ -43,16 +44,16 @@ var waitForDeletion = &metav1.DeleteOptions{
 func TestAutoscalerBasicFunctions(t *testing.T) {
 	t.Parallel()
 
-	alpha1 := framework.AgonesClient.StableV1alpha1()
-	fleets := alpha1.Fleets(defaultNs)
+	stable := framework.AgonesClient.StableV1alpha1()
```
Comment (pooneh-m) [inline test/e2e/fleetautoscaler_test.go]: autoscaling := framework.AgonesClient.AutoscalingV1alpha1()

File: test/e2e/fleetautoscaler_test.go (hunk 7 above)
Comment (markmandel) [reply test/e2e/fleetautoscaler_test.go]: yeah, that's a fair point :+1: I'll jump on it.

Right now it's a bit "v1alpha1" is the import if that's the most specific thing you are working on (for better or worse), and I think it's mostly like that in a lot of the codebase.

Comment (pooneh-m) [reply test/e2e/fleetautoscaler_test.go]: I'll merge this change to make sure this gets in for v0.11 release. We can fix this in future PRs.

**Pull request merged**